Adds SAM Environment

clomeli-ms opened the pull request:
Adds data-labeling-sam curated environment.

clomeli-ms commented:
Need to fix some things

vizhur commented on assets/data-labeling/environments/data-labeling-sam/context/requirements.txt:
```diff
@@ -0,0 +1,19 @@
+inference-schema~=1.5.0
+regex
+opencv-python
+pycocotools
```
better have this from conda to avoid build

vizhur commented on assets/data-labeling/environments/data-labeling-sam/context/requirements.txt:
```diff
@@ -0,0 +1,19 @@
+pycocotools
+matplotlib
+onnxruntime
+onnx
```
base image might have onnx runtime

vizhur commented on assets/data-labeling/environments/data-labeling-sam/context/requirements.txt:
```diff
@@ -0,0 +1,19 @@
+azureml-core==1.51.0
```
I'd resolve these to the latest from pypi to minimize maintenance.

vizhur commented on assets/data-labeling/environments/data-labeling-sam/context/Dockerfile:
```diff
@@ -0,0 +1,20 @@
+RUN pip install -r requirements.txt --no-cache-dir
+
+# Inference requirements
+COPY --from=mcr.microsoft.com/azureml/o16n-base/python-assets:20230419.v1 /artifacts /var/
```
do you need this or just copied from some env?

clomeli-ms replied:
just copied, will remove.

vizhur commented on assets/data-labeling/environments/data-labeling-sam/context/Dockerfile:
```diff
@@ -0,0 +1,20 @@
+FROM mcr.microsoft.com/azureml/aifx/stable-ubuntu2004-cu117-py38-torch1131:biweekly.202305.3
```
use a template to resolve latest base image. base image from May will result a sev2 for your team the moment you release your environment

elesel replied:
See https://github.com/Azure/azureml-assets/wiki/Environments#template-tags-1

clomeli-ms commented:
I believe I addressed all the comments left previously. Still seeing there is an issue in tests related to Name 'data-labeling-sam' is missing framework. I have made the PR into mcr btw: https://github.com/microsoft/mcr/pull/2691

elesel commented on assets/data-labeling/environments/data-labeling-sam/asset.yaml:
Please add one or more categories. See https://github.com/Azure/azureml-assets/wiki/Assets#categories

elesel commented on assets/data-labeling/environments/data-labeling-sam/context/conda_dependencies.yaml:
```diff
@@ -0,0 +1,28 @@
+  - azureml-contrib-services==1.51.0
+  - azureml-contrib-services==1.51.0
+  - torch-tb-profiler~=0.4.0
+  - azureml-automl-common-tools==1.51.0
```
Looks like this needs to change to 1.52.0. See the [build logs](https://github.com/Azure/azureml-assets/actions/runs/5664653344/job/15348256628?pr=925#step:7:515).

vizhur commented on assets/data-labeling/environments/data-labeling-sam/context/conda_dependencies.yaml:
```diff
@@ -0,0 +1,28 @@
+  - azureml-contrib-dataset=={{latest-pypi-version}}
+  - azureml-dataset-runtime[fuse,pandas]
+  - azureml-defaults=={{latest-pypi-version}}
+  - azure-ml==0.0.1
```
this one is not supported

vizhur commented on assets/data-labeling/environments/data-labeling-sam/context/conda_dependencies.yaml:
```diff
@@ -0,0 +1,28 @@
+  - azureml-defaults=={{latest-pypi-version}}
+  - azure-ml==0.0.1
+  - azure-ml-component==0.9.18.post2
+  - azureml-mlflow==1.51.0
```
all azureml- packages from common release should be pinned to the same version
